File: 3.py
#!/usr/bin/python
# -*- coding: UTF-8 -*-


def find(array, number):
    if array == None or number == None: # 是否为有效参数
        return False
    rows = len(array)
    columns = len(array[0])
    j = columns - 1
    i = 0
    # 从右上角开始查找目标值
    while j >=0 and i < rows:
        if array[i][j] == number:
            return True
        elif array[i][j] < number:
            i += 1
        else:
            j -= 1

    


if __name__ == "__main__":
    ll = [[1,2,8,9], [2,4,9,12], [4,7,10,13], [6,8,11,15]]
    print(find(ll, 7))

File: 3.cpp
#include <iostream>

using namespace std;

bool find(int* matrix, int rows, int columns, int target) {
    bool flag = false;
    if (matrix == NULL or rows < 0 or columns < 0) return flag;
    int row = 0;
    int column = columns - 1;
    while (column >= 0 && row < rows) {
        if (matrix[row*columns + column] == target) {
            flag = true;
            break;
        } else if (matrix[row*columns + column] < target) {
            ++row;
        } else {
            --column;
        }
    }
    return flag;
}

int main()
{
    int matrix[16] = {1,2,8,9,2,4,9,12,4,7,10,13,6,8,11,15};
    cout << "result:" << find(matrix, 4, 4, 1) << endl;
    return 0;
}